mava/components/jax/training/trainer.py

```diff
@@ -15,57 +15,89 @@
 
 from mava.components.jax import Component
 from mava.core_jax import SystemBuilder, SystemTrainer
 from mava.utils import enums
 from mava.utils.sort_utils import sort_str_num
 
 
+class BaseTrainerInit(Component):
+    @abc.abstractmethod
+    def __init__(
+        self,
+        config: Any,
+    ):
+        """Initialise system init components.
+
+        Args:
+            config : a dataclass specifying the component parameters.
+        """
+        self.config = config
+
+    @abc.abstractmethod
+    def on_building_init_end(self, builder: SystemBuilder) -> None:
+        """Summary."""
+        pass
+
+    @abc.abstractmethod
+    def on_training_utility_fns(self, trainer: SystemTrainer) -> None:
+        """Summary."""
+        pass
+
+    @staticmethod
+    def name() -> str:
+        """Component name."""
+
+        return "trainer_init"
+
+
 @dataclass
-class TrainerInitConfig:
+class BaseTrainerInitConfig:
     trainer_networks: Union[
         Dict[str, List], enums.Trainer
     ] = enums.Trainer.single_trainer
 
 
-class TrainerInit(Component):
-    def __init__(self, config: TrainerInitConfig = TrainerInitConfig()):
-        """_summary_
+@dataclass
+class SingleTrainerInitConfig(BaseTrainerInitConfig):
+    pass
+
+
+class SingleTrainerInit(BaseTrainerInit):
+    def __init__(self, config: SingleTrainerInitConfig = SingleTrainerInitConfig()):
+        """Initialises a single trainer.
+
+        Here a single trainer is used to train all networks.
 
         Args:
-            config : _description_.
+            config : a dataclass specifying the component parameters.
         """
         self.config = config
 
     def on_building_init_end(self, builder: SystemBuilder) -> None:
-        """_summary_
+        """Assigns trainers to networks for training.
 
         Args:
-            builder : _description_
+            builder : the system builder
         Raises:
-            ValueError: _description_
+            ValueError: Raises an error when trainer_networks is not
+                        set to single_trainer.
         """
         trainer_networks = self.config.trainer_networks
         unique_net_keys = builder.store.unique_net_keys
 
         # Setup trainer_networks
-        if not isinstance(trainer_networks, dict):
-            if trainer_networks == enums.Trainer.single_trainer:
-                builder.store.trainer_networks = {"trainer": unique_net_keys}
-            elif trainer_networks == enums.Trainer.one_trainer_per_network:
-                builder.store.trainer_networks = {
-                    f"trainer_{i}": [unique_net_keys[i]]
-                    for i in range(len(unique_net_keys))
-                }
-            else:
-                raise ValueError(
-                    "trainer_networks does not support this enums setting."
-                )
-        else:
-            builder.store.trainer_networks = trainer_networks
+
+        if trainer_networks != enums.Trainer.single_trainer:
```

Comment:
I cannot see any use in the JAX systems. I think you should create an issue to remove enums once we deprecate TF and maybe add a TODO comment here also

Reply:
The components do not require any enums any more, but I will still make an issue for investigating the use of enums after deprecating tf.